src/main/java/com/xiaomi/infra/pegasus/rpc/async/TableHandler.java

```diff
@@ -118,22 +119,36 @@ void initTableConfiguration(query_cfg_response resp) {
       newConfig.replicas.add(newReplicaConfig);
     }
 
-    // set partition configuration by resp, and create sessions
+    // create sessions for primary and secondaries
     FutureGroup<Void> futureGroup = new FutureGroup<>(resp.getPartition_count());
     for (partition_configuration pc : resp.getPartitions()) {
       ReplicaConfiguration s = newConfig.replicas.get(pc.getPid().get_pidx());
       s.ballot = pc.ballot;
-      s.primary = pc.primary;
-      if (pc.primary.isInvalid()) {
-        s.session = null;
-      } else {
-        if (s.session == null || !s.session.getAddress().equals(pc.primary)) {
-          // reset to new primary
-          s.session = manager_.getReplicaSession(pc.primary);
-          ChannelFuture fut = s.session.tryConnect();
-          if (fut != null) {
-            futureGroup.add(fut);
-          }
+
+      // create primary sessions
+      s.primaryAddress = pc.primary;
+      if (!pc.primary.isInvalid()) {
+        s.primarySession = manager_.getReplicaSession(pc.primary);
+        ChannelFuture fut = s.primarySession.tryConnect();
+        if (fut != null) {
+          futureGroup.add(fut);
+        }
+
+        // backup request is enabled, get all secondary sessions
+        s.secondarySessions.clear();
+        if (manager_.isEnableBackupRequest()) {
+          // secondary sessions
+          pc.secondaries.forEach(
+              secondary -> {
+                if (!secondary.isInvalid()) {
+                  ReplicaSession session = manager_.getReplicaSession(secondary);
+                  s.secondarySessions.add(session);
+                  ChannelFuture channelFuture = session.tryConnect();
+                  if (channelFuture != null) {
+                    futureGroup.add(channelFuture);
+                  }
+                }
+              });
         }
```

Comment:
`primarySessons` and `secondarySessions` actually is same node, so you shouldn't reconnect.

Reply:
What did you mean about "same" node? primary and secondary must be not in a same server, in consideration of high avialable.

Reply:
> same node

Oh, I didn't make it clear. actually is  `the same node-ip connection`. you only need connect all node, the session can be used for primary and secondary.

Reply:
Well, I see